sys/hacl/build.rs

```diff
@@ -95,92 +117,147 @@ fn copy_files(home_path: &Path, out_path: &Path) {
     fs_extra::dir::copy(home_path.join("c"), out_path, &options).unwrap();
 }
 
-fn compile_files(files: &[String], out_path: &Path, args: &[String]) {
-    let mut clang_args = includes(out_path);
-
-    clang_args.push("-O3".to_string());
-    clang_args.push("-c".to_string());
-    clang_args.extend_from_slice(args);
-
-    let mut build_cmd = Command::new("clang");
-    let mut build_args = clang_args;
-    build_args.extend_from_slice(files);
-    println!(" >>> {}", out_path.join("c").join("src").display());
-    println!(" >>> {}", build_args.join(" "));
-    build_cmd
-        .current_dir(out_path.join("c").join("src"))
-        .args(&build_args);
-    println!(" >>> build_cmd: {:?}", build_cmd);
-    println!("     current dir: {:?}", build_cmd.get_current_dir());
-    let build_status = build_cmd.status().expect("Failed to run build.");
-    println!(" >>> build status: {:?}", build_status);
-    println!(" >>> out {:?}", out_path);
-    assert!(build_status.success());
+fn compile_files(
+    library_name: &str,
+    files: &[String],
+    out_path: &Path,
+    args: &[String],
+    defines: &[(&str, &str)],
+) {
+    let src_prefix = out_path.join("c").join("src");
+
+    let mut build = cc::Build::new();
+    build
+        .files(files.iter().map(|fname| src_prefix.join(fname)))
+        // XXX: There are too many warnings for now
+        // .warnings_into_errors(true)
+        .warnings(true)
+        .extra_warnings(true);
+    // .no_default_flags(true);
+
+    for include in includes(out_path, "") {
+        build.include(include);
+    }
+    build.flag("-O3").flag("-c");
```

Comment:
I have to check this later but I expect that this will fail on Windows when compiling with MSVC.
Let's use the `opt_level` function instead. The `-c` we don't need at all I think.

Reply:
Done~